src/registry.rs

```diff
@@ -77,6 +77,14 @@ impl<M> Default for Registry<M> {
 }
 
 impl<M> Registry<M> {
+    /// Creates a new default registry with the given prefix.
+    pub fn with_prefix(prefix: impl Into<String>) -> Self {
```

Comment:
```suggestion
    pub fn new_with_prefix(prefix: impl Into<String>) -> Self {
```

This would be more intuitive for me. What do you think @Victor-N-Suadicani ?

Reply:
The convention in Rust is to just use `with_something` - check for instance `with_capacity` on `Vec` or `with_hasher` on `HashMap` or just search the `std` docs for `with_` for lots of examples.